public/class-cf7-grid-layout-public.php

```diff
@@ -255,12 +256,28 @@ public function cf7_shortcode_request($output, $tag, $attr){
       wp_enqueue_script( $cf7_key.'-js' , $themeuri.'/js/'.$cf7_key.'.js', array($this->plugin_name), null, true);
       do_action('smart_grid_register_custom_script', $cf7_key);
     }
+	  
+    // enable select2 script when the keyword "select2" (in classNames) occurs in $output
+    $class['has-select2'] = strpos($output, 'select2') !== false;    
```

Comment:
this would make it slow for large forms several 100s of html lines.  this logic needs to be handled in the admin side, at teh time of saving the form, possibly even on the admin client side using js prior to submission.